driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/dup.bpf.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(dup_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, DUP_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_DUP_1_E, DUP_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: oldfd (type: PT_FD) */
+	s32 oldfd = (s32)extract__syscall_argument(regs, 0);
+	ringbuf__store_s64(&ringbuf, (s64)oldfd);
```

Comment:
I feel like this casting from `s32` to `s64` could lead to garbage data being pushed into the ringbuffer, the compiler might be handling this correctly by zeroing out the top bits of the variable, but it still makes me a bit unneasy.

Reply:
Uhm to be honest, I expect it to be correctly converted into 64 bit, otherwise, there will be something wrong in the compiler, but I have no strong opinion on that, what would you propose instead of this approach? :thinking:

Reply:
Mmmh i think i agree with Andrea: we are casting a signed 32b to a signed 64b before passing it by value to the `ringbuf__store_s64`.
I think we could also directly pass the s32 and it would be upcasted right in any case.

Reply:
Alright, I'll trust you guys trusting the compiler, I've just always been extremely paranoid with these types of statements.